Features/AvatarToolkit/Sources/Scenes/EmojiStyleScene.swift

```diff
@@ -0,0 +1,173 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import Foundation
+import SwiftUI
+import Style
+import Localization
+
+public struct EmojiStyleScene: View {
+    private enum Tab: Equatable {
+        case emoji, style
+    }
+
+    @Environment(\.dismiss) private var dismiss
+    
+    private let emojiViewSize = Sizing.image.extraLarge
+    @State private var selectedTab: Tab = .emoji
+    @Binding private var image: UIImage?
+    
+    @State private var model = EmojiViewModel()
+
+    public init(image: Binding<UIImage?>) {
+        _image = image
+    }
+    
+    public var body: some View {
+        NavigationStack {
+            VStack {
+                emojiView(color: model.color, emoji: model.text)
+                
+                changeBackgroundButton
+
+                picker
+                    .padding(.horizontal, Spacing.medium)
+                    .padding(.top, Spacing.medium)
+
+                Group {
+                    switch selectedTab {
+                    case .emoji:
+                        emojiListView
+                    case .style:
+                        colorsView
+                    }
+                }
+                .padding(.horizontal, Spacing.medium)
+                .padding(.top, Spacing.medium)
+                
+                magicButton
+                    .padding(.horizontal, Spacing.medium)
+
+                Spacer()
+            }
+            .toolbar {
+                ToolbarItem(placement: .navigationBarLeading) {
+                    Button(Localized.Common.cancel) {
+                        dismiss()
+                    }.bold()
+                }
+                ToolbarItem(placement: .navigationBarTrailing) {
+                    Button(Localized.Common.done) {
+                        image = emojiView(color: model.color, emoji: model.text)
+                            .render(for: CGSize(width: emojiViewSize, height: emojiViewSize))
+                        dismiss()
+                    }
+                    .bold()
+                    .disabled(model.text.isEmpty)
+                }
+            }
+        }
+    }
+    
+    private func emojiView(color: Color, emoji: String) -> some View {
+        EmojiView(color: color, emoji: emoji)
+            .frame(width: emojiViewSize, height: emojiViewSize)
+    }
+    
+    private var picker: some View {
+        Picker("", selection: $selectedTab.animation()) {
+            Text(Localized.Common.emoji).tag(Tab.emoji)
+            Text(Localized.Common.style).tag(Tab.style)
+        }
+        .pickerStyle(SegmentedPickerStyle())
+    }
+    
+    private var changeBackgroundButton: some View {
+        Button(action: {
+            model.color = .random()
+        }) {
+            Text(Localized.Avatar.changeBackground + Emoji.WalletAvatar.magic.rawValue)
```

Comment:
```suggestion
            Text(model....)
```

model should return value